backend/fms_core/models/sample.py

```diff
@@ -264,65 +127,27 @@ def add_error(field: str, error: str):
 
         self.normalize()
 
-        # Sample Kind
-        sample_kind_choices = (Sample.BIOSPECIMEN_TYPE_NA_CHOICES if self.extracted_from
-                               else Sample.BIOSPECIMEN_TYPE_CHOICES)
-
-        if self.sample_kind_id is not None:
-            if self.sample_kind.name not in frozenset(c[0] for c in sample_kind_choices):
-                add_error(
-                    "sample_kind",
-                    (f"Sample Kind name {self.sample_kind.name} not valid for "
-                     f"{' extracted' if self.extracted_from else ''} sample {self.name}"),
-                )
-            # Check concentration fields given sample_kind
-            if self.sample_kind.name in Sample.BIOSPECIMEN_TYPES_CONC_REQUIRED and self.concentration is None:
-                add_error("concentration", "Concentration must be specified if the sample_kind is DNA")
-
-            # Check tissue source given extracted_from
-
-            if self.tissue_source and self.sample_kind.name not in Sample.BIOSPECIMEN_TYPES_NA:
-                add_error("tissue_source", "Tissue source can only be specified for a nucleic acid sample.")
-
-            if self.transferred_from and self.sample_kind != self.transferred_from.sample_kind:
-                    add_error("sample_kind", "Sample kind need to remain the same during transfer")
+        DerivedSample = apps.get_model("fms_core", "DerivedSample")
+        # Check concentration fields given sample_kind
+        if self.concentration is None and (self.is_pool or self.derived_samples.objects.first().sample_kind.name in DerivedSample.BIOSPECIMEN_TYPES_CONC_REQUIRED):
+            add_error("concentration", "Concentration must be specified for a pool or if the sample_kind is DNA")
 
-            if self.extracted_from and self.sample_kind.name not in Sample.BIOSPECIMEN_TYPES_NA:
-                add_error("sample_kind", "Extracted sample need to be a type of Nucleic Acid.")
-
-
-        if self.extracted_from:
-            if self.extracted_from.sample_kind.name in Sample.BIOSPECIMEN_TYPES_NA:
-                add_error(
-                    "extracted_from",
-                    f"Extraction process cannot be run on sample of type {', '.join(Sample.BIOSPECIMEN_TYPES_NA)}"
-                )
-
-            original_sample_kind = self.extracted_from.sample_kind.name
-            if self.tissue_source != Sample.BIOSPECIMEN_TYPE_TO_TISSUE_SOURCE[original_sample_kind]:
-                add_error(
-                    "tissue_source",
-                    (f"Mismatch between sample tissue source {self.tissue_source} and original sample kind "
-                     f"{original_sample_kind}")
-                )
+        if self.transferred_from:
+            if list(self.derived_samples.objects.all()) != list(self.transferred_from.derived_samples.objects.all()):
```

Comment:
I'd suggest comparing an unordered (or sorted) list of ids. Not sure that you can compare a django queryset transformed to a python list that easily. Would the order matter? You would be comparing a list of python objects I think

Reply:
We'll have to test it but i assume the list will be ordered by id since i think by default queryset orders using the primary key.

Reply:
It might work, but in all cases converting a queryset to a python list, and comparing the two lists that are filled with objects... i am not sure about the soundness of that solution. at least there is probably a function or an operator for list comparison in python

Reply:
yeah i am uncertain too but i would like to try it ... a list of objects should be easily enough comparable to another list of object. Each object should be equal if their primary key is identical ... it is a much clearer simpler way to to this than to build 2 list of ids in manually then comparing them ... i think in the end it's likely doing the same...